Assets/BossRoom/Scripts/Server/Game/Character/ServerCharacter.cs

```diff
@@ -214,6 +220,33 @@ public float GetBuffedValue(Action.BuffableValue buffType)
             return m_ActionPlayer.GetBuffedValue(buffType);
         }
 
+        /// <summary>
+        /// Returns the range at which this character can detect enemies. Only applicable to NPCs.
+        /// This is usually the same value as is indicated by our game data, but it can be
+        /// dynamically overridden.
+        /// </summary>
+        /// <returns>our entity detection range (in meters)</returns>
+        public float GetDetectRadius()
+        {
+            if (m_DetectRadiusOverride == -1)
+            {
+                return GameDataSource.Instance.CharacterDataByType[NetState.CharacterType].DetectRange;
+            }
+            else
+            {
+                return m_DetectRadiusOverride;
+            }
+        }
+
+        /// <summary>
+        /// Changes this creature's detection radius. Only meaningful for NPCs.
+        /// </summary>
+        /// <param name="radius">new detection radius for this entity (in meters)</param>
+        public void SetDetectRadius(float radius)
+        {
+            m_DetectRadiusOverride = radius;
+        }
+
```

Comment:
Could this be placed someplace else that's pertinent only to NPCs that have detecting capabilities? This ServerCharacter component shouldn't be aware of this, I feel.

Reply:
I think it belongs in `AIBrain`, but because that's a private object owned by `ServerCharacter` (and not a component) I felt weird exposing it. Here's what that looks like with `ServerCharacter` providing an `AIBrain` accessor, and `AIBrain` having the radius.

I think we had discussed making AIBrain its own component eventually, and that's an existing jira task, so I'm fine with this approach given that it will likely become a component in the future

Reply:
i'm okay with that for now!

Reply:
This works great for now.